### PR TITLE
Split item filtering into separate source file

### PR DESCRIPTION
**Pull request by dedmen**
Addition onto #57
Merge current Master branch (would be better to do that into rework branch first)
Split the item indexing/filtering out into separate files, and fix the broken conditions introduced in #57

Actual changes in https://github.com/azerothcore/mod-ah-bot/pull/75/commits/eab81cf0c360b7ded55197eb5f64a654de8c5f84

I compared item counts from before(non-broken master branch)/after to confirm they match

```
AuctionHouseBot:
5957 disabled items
Loaded 5 grey trade goods
Loaded 471 white trade goods
Loaded 58 green trade goods
Loaded 26 blue trade goods
Loaded 3 purple trade goods
Loaded 1 orange trade goods
Loaded 0 yellow trade goods
Loaded 1633 grey items
Loaded 2771 white items
Loaded 5520 green items
Loaded 1645 blue items
Loaded 809 purple items
Loaded 0 orange items
Loaded 1 yellow items
```

With Rogue items disabled (Because I modified the way disabled class items are filtered)

```
AuctionHouseBot:
5957 disabled items
Loaded 5 grey trade goods
Loaded 471 white trade goods
Loaded 58 green trade goods
Loaded 26 blue trade goods
Loaded 3 purple trade goods
Loaded 1 orange trade goods
Loaded 0 yellow trade goods
Loaded 1628 grey items
Loaded 2699 white items
Loaded 5520 green items
Loaded 1645 blue items
Loaded 807 purple items
Loaded 0 orange items
Loaded 1 yellow items
```

All still matches

### CODE REVIEW

**Review by Helias — APPROVED** [review]

**Comment by Helias** [comment]
thanks for your PR!
Build is failing, check the errors in the pipeline please

File: src/AuctionHouseBot.cpp [hunk 1]
```diff
@@ -120,6 +106,8 @@ void AuctionHouseBot::AddNewAuctions(Player* AHBplayer, AHBConfig* config)
 
     LOG_DEBUG("module.ahbot", "AHSeller: {} items", items);
 
+    const auto itemIndex = sAHIndex;
```
Comment (Winfidonarleyan) [inline src/AuctionHouseBot.cpp:109]:
```suggestion
    auto const itemIndex = sAHIndex;
```

**Comment by dedmen** [comment]
Buildfail is actually something from rework branch
`AuctionHouseBot.h:3:1: fatal error: '/*' within block comment [-Wcomment]`


Also I put the wrong license header, trinitycore one instead of AC

**Comment by dedmen** [comment]
Are you okey with my file naming? The other files all prefix "AuctionHouseBot", but I don't think there is actually a technical reason to do that?
I would prefer shorter names, we already know this is AuctionHouseBot because it's in "mod-ah-bot/src" folder

**Comment by dedmen** [comment]
This is still waiting, any update?

I would like to make more contributions but its very hard to handle all the conflicts.

**Comment by Helias** [comment]
merged, thanks!

**Pull request merged**